service/service_test.go

```diff
@@ -0,0 +1,436 @@
+			service.GitCommit = "TestGitCommit"
+			service.Version = "TestVersion"
+
+			svcErrors := make(chan error, 1)
+
+			cfg, err := config.Get()
+			So(err, ShouldBeNil)
+
+			svc := &service.Service{
+				Config:      cfg,
+				HealthCheck: hcMock,
+				Server:      failingServerMock,
+				ServiceList: mockServiceList,
+			}
+
+			Convey("When service starts", func() {
+				svc.Start(ctx, svcErrors)
+
+				Convey("Then service start fails and returns an error in the error channel", func() {
+					sErr := <-svcErrors
+					So(sErr.Error(), ShouldResemble, errServer.Error())
+					So(len(failingServerMock.ListenAndServeCalls()), ShouldEqual, 1)
+				})
+			})
+		})
+	})
+}
+
+func TestCloseSuccess(t *testing.T) {
+	Convey("Given a correctly initialised service", t, func() {
+
+		ctx := context.Background()
+
+		cfg, err := config.Get()
+		So(err, ShouldBeNil)
+
+		hcStopped := false
+
+		// healthcheck Stop does not depend on any other service being closed/stopped
+		hcCloseMock := &mocks.HealthCheckerMock{
+			AddCheckFunc: func(name string, checker healthcheck.Checker) error { return nil },
+			StartFunc:    func(ctx context.Context) {},
+			StopFunc:     func() { hcStopped = true },
+		}
+
+		// server Shutdown will fail if healthcheck is not stopped
+		serverCloseMock := &mocks.HTTPServerMock{
+			ListenAndServeFunc: func() error { return nil },
+			ShutdownFunc: func(ctx context.Context) error {
+				if !hcStopped {
+					return errors.New("Server stopped before healthcheck")
+				}
+				return nil
+			},
+		}
+
+		serviceList := service.NewServiceList(nil)
+		serviceList.HealthCheck = true
+		svc := service.Service{
+			Config:      cfg,
+			HealthCheck: hcCloseMock,
+			Server:      serverCloseMock,
+			ServiceList: serviceList,
+		}
+
+		Convey("When closing service", func() {
+			err = svc.Close(ctx)
+
+			Convey("Then it results in all the dependencies being closed in the expected order", func() {
+				So(err, ShouldBeNil)
+				So(len(hcCloseMock.StopCalls()), ShouldEqual, 1)
+				So(len(serverCloseMock.ShutdownCalls()), ShouldEqual, 1)
+			})
+		})
+	})
+}
+
+func TestCloseFailure(t *testing.T) {
+	Convey("Given if service fails to stop", t, func() {
+		failingServerCloseMock := &mocks.HTTPServerMock{
+			ListenAndServeFunc: func() error { return nil },
+			ShutdownFunc: func(ctx context.Context) error {
+				return errors.New("Failed to stop http server")
+			},
+		}
+
+		Convey("And given a correctly initialised service", func() {
+			ctx := context.Background()
+
+			cfg, err := config.Get()
+			So(err, ShouldBeNil)
+
+			serviceList := service.NewServiceList(nil)
+			serviceList.HealthCheck = true
+			svc := service.Service{
+				Config:      cfg,
+				HealthCheck: hcMock,
+				Server:      failingServerCloseMock,
+				ServiceList: serviceList,
+			}
+
+			Convey("When closing the service", func() {
+				err = svc.Close(ctx)
+
+				Convey("Then Close operation tries to close all dependencies and returns an error", func() {
```

Comment:
The "and returns an error" part is done in the following Convey so should probably be removed from this one
```suggestion
				Convey("Then Close operation tries to close all dependencies", func() {
```